src/libstd/io/mod.rs

```diff
@@ -1444,6 +1444,13 @@ pub struct Chain<T, U> {
     done_first: bool,
 }
 
+#[stable(feature = "std-debug", since = "")]
+impl<T, U> fmt::Debug for Chain<T, U> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad("Chain {{ .. }}")
```

Comment:
IMO the IO adaptors should print their inner reader/writers. They have nice debug output too.